src/rbindlist.c

```diff
@@ -305,14 +308,25 @@ SEXP rbindlist(SEXP l, SEXP usenamesArg, SEXP fillArg, SEXP idcolArg)
       if (INHERITS(thisCol, char_integer64)) {
         if (firsti>=0 && !length(getAttrib(firstCol, R_ClassSymbol))) { firsti=i; firstw=w; firstCol=thisCol; } // so the integer64 attribute gets copied to target below
         int64=true;
+      } else if (INHERITS(thisCol, char_Date))  {
+        if (firsti>=0 && !length(getAttrib(firstCol, R_ClassSymbol))) { firsti=i; firstw=w; firstCol=thisCol; }
```

Comment:
we don't need to `PROTECT()` the `getAttrib()` result because it's not stored, right?

Reply:
I think we do not have to `PROTECT` since it is not allocating something new but only a getter.
And they never use `PROTECT` for getAttrib in WRE

Reply:
We do use `PROTECT(getAttrib())` in a few places, so I've marked this as something to look into in #6280. Will be easier after #6263